extensions/blocks/simple-payments/save.js

```diff
@@ -1,9 +1,60 @@
 /**
- * External dependencies
+ * WordPress dependencies
  */
-import { RawHTML } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { formatPrice } from './utils';
 
 export default function Save( { attributes } ) {
-	const { productId } = attributes;
-	return productId ? <RawHTML>{ `[simple-payment id="${ productId }"]` }</RawHTML> : null;
+	const {
+		content,
+		currency,
+		featuredMediaUrl,
+		featuredMediaTitle,
+		postLink,
+		price,
+		productId,
+		title,
+	} = attributes;
+	if ( ! productId ) {
+		return null;
+	}
+
+	return (
+		<div className={ `jetpack-simple-payments-wrapper jetpack-simple-payments-${ productId }` }>
+			<div className="jetpack-simple-payments-product">
+				{ featuredMediaUrl && (
+					<div className="jetpack-simple-payments-product-image">
+						<div className="jetpack-simple-payments-image">
+							<figure>
+								<img src={ featuredMediaUrl } alt={ featuredMediaTitle } />
+							</figure>
+						</div>
+					</div>
+				) }
+				<div className="jetpack-simple-payments-details">
+					<div className="jetpack-simple-payments-title">
+						<p>{ title }</p>
+					</div>
+					<div className="jetpack-simple-payments-description">
+						<p>{ content }</p>
+					</div>
+					<div className="jetpack-simple-payments-price">
+						<p>{ formatPrice( price, currency ) }</p>
+					</div>
+					<a
+						className="jetpack-simple-payments-purchase"
+						href={ postLink }
+						target="_blank"
+						rel="noopener noreferrer"
+					>
+						{ __( 'Visit the site to purchase.', 'jetpack' ) }
```

Comment:
I wonder if this may cause some block validation issues in some cases, as we addressed with other blocks in #16361?

Reply:
It can certainly causes the block invalidation. Handled in 1dfc48f.